cmd/zed/zed.c

```diff
@@ -204,6 +204,8 @@ _finish_daemonize(void)
 
 	/* Notify parent that daemonization is complete. */
 	zed_log_pipe_close_writes();
+
+	(void) close(devnull);
```

Comment:
This isn't quite right.  `devnull` will under normal conditions be larger than `STDERR_FILENO` so this will be closed on line 202.  If that's somehow not that case then one of the `dup2()` call above must fail.  So this actually looks like a false positive to me since Coverity probably doesn't understand the constraints on `STD*_FILENO`'s.  Plus adding the close here would result in `devnull` being closed twice.

Reply:
Thanks @behlendorf. I'll try to be more careful. Thanks for you patience.

Reply:
It looks like you forgot to drop this hunk from the patch.

Reply:
Ouch, I swear I was not being ironic in the comment above.